### ISSUE TITLE
new superFunction inconsistency in v1.5.0

### ISSUE THREAD

**Issue by dagda1** [emberjs/ember.js#4632]
I've just upgraded to ember.js 1.5.0 and the way the new ```superFunction``` works is breaking some existing code.

I have recreated the problem in this [jsbin](http://jsbin.com/vibeg/6/edit)

To recreate
1. Click on any of the links which will render a modal dialogue template into an outlet named modal.
2. Click close
When close is clicked this action handler is executed:
```
closeModal: function() {
 this.render('nothing', {
  into: 'application',
  outlet: 'modal'
 });
```
Which causes the modal view's destroy handler to be executed:
```
destroy: function() {
  var superFunction;
  if (this.get('state') !== 'inDOM') {
     this._super();
  }
  this.$('.modal').removeClass('in');
  superFunction = this._super.bind(this);
  return this.$().one('webkitTransitionEnd', superFunction);
}
```
The problem is that when the ```webkitTransitionEnd``` event is fired, ```this.this.__nextSuper``` is undefined in the ```superFunction```.

I can work round this by creating a copy of ```__nextSuper``` like this before destroy finishes:
```
destroy: function() {
  var superFunction;
  if (this.get('state') !== 'inDOM') {
     this._super();
  }
  this.$('.modal').removeClass('in');
  superFunction = this.__nextSuper.bind(this);
  return this.$().one('webkitTransitionEnd', superFunction);
}
```
Is this behaviour what you would expect or a bug?

**Comment by stefanpenner** [comment]
Super + async wont be solved until we migrate to full ES6 classes.

Luckily a very simple workaround exists (more or less what the ES6 -> ES5 transpiler does):

```
export default Ember.Controller.extend({
  something() {
   var boundSuper = this._super.bind(this);

    ajax().then(function() {
      boundSuper(...args);
    });
  }
});
```

**Comment by Fryie** [comment]
@stefanpenner I have this weird problem where that workaround won't work in my app, but when I tried making a twiddle, it did, so it must be something very specific. When I debug the call, I end at some code like this:
```
function superFunction() {
    var func = this.__nextSuper;
    var ret;

    if (func) {
       ...
    }
}
```

Only trouble is, `this.__nextSuper` ends up being undefined, even with your workaround. Any idea? I'm on 1.13, so maybe this has been fixed in a more recent version (hence why it works on the twiddle). FWIW, I got it to work by calling ```MyMixin.mixins[0].properties.mySuperFunction.call(MyMixin.mixins[0])``` which is incredibly dirty.
